docs/style-guide.rst

```diff
@@ -0,0 +1,162 @@
+
+    * **Modules** have short, all-lowercase names. Underscores can be used in the module name if it improves readability.
+    * **Class names** use the CapWords convention.
+    * **Exceptions** follow the same conventions as other classes.
+    * **Function** names are lowercase, with words separated by underscores when it improves readability.
+    * **Method** names and **instance** variables follow the same conventions as functions.
+    * **Constants** use all capital letters with underscores separating words.
+
+Booleans
+********
+
+1. Boolean values **should** be prefixed with ``is_``.
+2. Booleans **must not** represent *negative* properties, (e.g. ``is_not_set``). This can result in double-negative evaluations which are not intuitive for readers.
+3. Methods that return a single boolean **should** use the :py:class:`@property<property>` decorator
+
+Methods
+*******
+
+We use the following conventions when naming methods:
+
+* ``get_*``: for data retrieval. **May** include validation. **Should not** have any side effects.
+* ``set_*``: for adding a new value or modifying an existing one. **May** include validation.
+* ``compare_*``: for comparing values. **Should** return ``True`` or ``False``. **Should not** raise an exception.
+* ``validate_*``: **Should** return ``None`` or raise an exception if something is wrong.
+* ``from_*``: for class methods that instantiate an object based on the given input data.
+
+For other functionality, try to choose names that clearly communicate intent without being overly verbose. Focus on *what* the method does, not on *how* it does it.
+
+Imports
+-------
+
+Import sequencing is handled with `isort <https://github.com/timothycrosley/isort>`_. We also follow these additional rules:
+
+1. Standard libraries **should not** be imported with ``from ...`` style syntax. Importing the library itself aids readability, as other users may be familiar with that library.
+2. Local imports **should** use absolute paths instead of relative. Relative imports cause issues if the module is moved.
+
+Exceptions
+----------
+
+Vyper uses :ref:`custom exception classes <compiler-exceptions>` to indicate what has gone wrong during compilation.
+
+1. All raised exceptions **must** use a exception class that appropriately describes that has gone wrong. When none fits, or we are using an exception class to cover an overly broad range of errors, consider creating a new class.
+2. Builtin python exceptions **must not** be intentionally raised. An unhandled builtin exception should indicate a bug in the codebase. Use :func:`CompilerPanic<CompilerPanic>` for errors that are not caused by the user.
+
+Strings
+-------
+
+Strings substitusions **should** be performed via `formatted string literals <https://docs.python.org/3/reference/lexical_analysis.html#formatted-string-literals>`_ rather than the ``str.format`` method or other techniques.
```

Comment:
```suggestion
Strings substitutions **should** be performed via `formatted string literals <https://docs.python.org/3/reference/lexical_analysis.html#formatted-string-literals>`_ rather than the ``str.format`` method or other techniques.
```